node/src/types/block.rs

```diff
@@ -1265,51 +1265,25 @@ impl Item for Block {
 
 /// A wrapper around `Block` for the purposes of fetching blocks by height in linear chain.
 #[derive(Clone, Debug, PartialOrd, Ord, PartialEq, Eq, Hash, Serialize, Deserialize)]
-pub enum BlockByHeight {
-    Absent(u64),
-    Block(Box<Block>),
+pub struct BlockWithMetadata {
```

Comment:
So you're saying now that there's never a need to download just block, without the finality signatures?

Reply:
I believe there _shouldn't_ be a need to download just a block by height, because it's insecure.